pkg/election/election.go

```diff
@@ -80,6 +101,32 @@ func NewElection(ctx context.Context, cli *clientv3.Client, sessionTTL int, key,
 		return nil, terror.ErrElectionCampaignFail.Delegate(err, "create the initial session")
 	}
 
+	for {
+		_, meta, err := e.LeaderInfo(ctx)
+		if err != nil {
+			return nil, errors.WithStack(err)
```

Comment:
We often use `errors.Trace(err)` and `errors.WithStack` is deprecated. or it's event better to add some terror in _pkg/terror/error_list.go_.